### PR TITLE
CSS Gather: Dockerize

### PR DESCRIPTION
**Pull request by mlahargou**
This adds a Dockerfile and docker-compose with the instructions on how to spin up this
app in a docker container. In the past, getting all of the different versions to work with 
each other has been a pain. This should help with that.

QA:
--
- Locally clone the repo.
- Run `docker-compose build`
- Run `docker run css-gather ./run.rb https://www.ifixit.com`
- The output should be written to stdout.

cc @sctice-ifixit @djmetzle @batbattur @ardelato 

Closes: https://github.com/ifixit/ifixit/issues/46405

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+COPY . ./
+
+COPY Gemfile Gemfile.lock ./
+
+RUN gem install bundler -v 2.4.6
+RUN bundle check || bundle install
+
+COPY package.json package-lock.json ./
+
+RUN npm install
```
Comment (djmetzle) [inline Dockerfile]: Super minor, if you reorder this a little bit, you'll get better caching of the package dependencies when iterating on the code:
```suggestion
WORKDIR /app

COPY Gemfile Gemfile.lock ./

RUN gem install bundler -v 2.4.6
RUN bundle check || bundle install

COPY package.json package-lock.json ./

RUN npm install

COPY . ./
```

Comment (mlahargou) [reply Dockerfile]: Sweet, sounds good to me!

**Review by djmetzle — APPROVED** [review]
I didnt test/QA that suggestion, but this looks good to me. Nice work @mlahargou! 
CR :gear:

**Comment by mlahargou** [comment]
> I didnt test/QA that suggestion 

I did minor QA just making sure that it still built. I'm assuming manual QA will do more thorough testing.

**Comment by deltuh-vee** [comment]
Might also want to update the Usage section in the README.

**Comment by mlahargou** [comment]
> Might also want to update the Usage section in the README.

I already have [a blocked issue to update that](https://github.com/iFixit/ifixit/issues/46407), but great callout! The old way will still work, so I'm not to concerned about updating the docs before we get this out.

**Comment by deltuh-vee** [comment]
dev_block ✊ 
I ran into an error during the last step of running `docker-compose build`
```
 => ERROR [ 9/10] RUN npm install                                                              32.8s
------
 > [ 9/10] RUN npm install:
#0 0.904 npm WARN npm npm does not support Node.js v10.24.0
#0 0.904 npm WARN npm You should probably upgrade to a newer version of node as we
#0 0.904 npm WARN npm can't make any promises that npm will work with this version.
#0 0.904 npm WARN npm Supported releases of Node.js are the latest release of 4, 6, 7, 8, 9.
#0 0.904 npm WARN npm You can find the latest version at https://nodejs.org/
#0 1.058 npm WARN read-shrinkwrap This version of npm is compatible with lockfileVersion@1, but package-lock.json was generated for lockfileVersion@2. I'll try to do my best with it!
#0 2.130 npm WARN notice [SECURITY] css-what has the following vulnerability: 1 high. Go here for more details: https://github.com/advisories?query=css-what - Run `npm i npm@latest -g` to upgrade your npm version, and then `npm audit` to get more info.
#0 2.414 npm WARN tar write after end
#0 2.514 npm WARN tar write after end
#0 2.939 npm WARN notice [SECURITY] postcss has the following vulnerability: 1 moderate. Go here for more details: https://github.com/advisories?query=postcss - Run `npm i npm@latest -g` to upgrade your npm version, and then `npm audit` to get more info.
#0 2.998 npm WARN tar write after end
#0 3.055 npm WARN tar write after end
#0 3.389 npm WARN tar write after end
#0 3.695 npm WARN notice [SECURITY] ws has the following vulnerability: 1 moderate. Go here for more details: https://github.com/advisories?query=ws - Run `npm i npm@latest -g` to upgrade your npm version, and then `npm audit` to get more info.
#0 3.759 npm WARN tar write after end
#0 4.715 npm WARN tar write after end
#0 4.817 npm WARN tar write after end
#0 5.427 npm WARN tar write after end
#0 5.511 npm WARN tar write after end
#0 5.829 npm WARN tar write after end
#0 5.894 npm WARN tar write after end
#0 5.943 npm WARN tar write after end
#0 6.283 npm WARN tar write after end
#0 6.583 npm WARN tar write after end
#0 7.533 npm WARN tar write after end
#0 7.780 npm WARN tar write after end
#0 8.576 npm WARN tar write after end
#0 9.195 npm WARN tar write after end
#0 9.351 npm WARN tar write after end
#0 9.584 npm WARN tar write after end
#0 9.738 npm WARN tar write after end
#0 9.899 npm WARN tar write after end
#0 10.18 npm WARN tar write after end
#0 10.50 npm WARN tar write after end
#0 10.61 npm WARN notice [SECURITY] nth-check has the following vulnerability: 1 high. Go here for more details: https://github.com/advisories?query=nth-check - Run `npm i npm@latest -g` to upgrade your npm version, and then `npm audit` to get more info.
#0 10.62 npm WARN notice [SECURITY] normalize-url has the following vulnerability: 1 high. Go here for more details: https://github.com/advisories?query=normalize-url - Run `npm i npm@latest -g` to upgrade your npm version, and then `npm audit` to get more info.
#0 10.79 npm WARN notice [SECURITY] browserslist has the following vulnerability: 1 moderate. Go here for more details: https://github.com/advisories?query=browserslist - Run `npm i npm@latest -g` to upgrade your npm version, and then `npm audit` to get more info.
#0 10.86 npm WARN notice [SECURITY] nanoid has the following vulnerability: 1 moderate. Go here for more details: https://github.com/advisories?query=nanoid - Run `npm i npm@latest -g` to upgrade your npm version, and then `npm audit` to get more info.
#0 10.87 npm WARN tar write after end
#0 11.13 npm WARN notice [SECURITY] minimatch has the following vulnerability: 1 high. Go here for more details: https://github.com/advisories?query=minimatch - Run `npm i npm@latest -g` to upgrade your npm version, and then `npm audit` to get more info.
#0 11.18 npm WARN tar write after end
#0 11.23 npm WARN tar write after end
#0 11.78 npm WARN tar write after end
#0 11.94 npm WARN tar write after end
#0 11.98 npm WARN tar write after end
#0 12.00 npm WARN tar write after end
#0 12.05 npm WARN tar write after end
#0 32.75 npm ERR! cb() never called!
#0 32.75
#0 32.75 npm ERR! This is an error with npm itself. Please report this error at:
#0 32.75 npm ERR!     <https://github.com/npm/npm/issues>
#0 32.77
#0 32.77 npm ERR! A complete log of this run can be found in:
#0 32.77 npm ERR!     /root/.npm/_logs/2023-02-08T22_34_19_168Z-debug.log
------
failed to solve: executor failed running [/bin/sh -c npm install]: exit code: 1
```

**Comment by mlahargou** [comment]
That's odd. It's working for all of qae it seems. I'll dig into it.

**Comment by ardelato** [comment]
@deltuh-vee was that ran on your `Framework` Windows machine?

**Comment by deltuh-vee** [comment]
> @deltuh-vee was that ran on your `Framework` Windows machine?

Yes it was.

**Comment by djmetzle** [comment]
Node 10.24? Thats a specific version. Try a few other versions? Hows 18?

**Comment by mlahargou** [comment]
@deltuh-vee Can you try that out? It's a newer version of the ruby image.

```
Node: v12.22.12
npm: 7.5.2
```

It still works on my machine.

**Comment by mlahargou** [comment]
I'm going to un_dev_block 📱 so we can try again. It's working for all of QAE and it's working in the github action: https://github.com/iFixit/ifixit/actions/runs/4129713894/jobs/7135623746

It _might_ be a problem with your windows machine. If it is, we can work on getting that fixed up. I made a change so hopefully it just works for you now.

**Comment by deltuh-vee** [comment]
QA 🎬 
`docker-compose build` and` docker run css-gather ./run.rb https://www.ifixit.com` can be successfully run.

**Review by djmetzle — APPROVED** [review]
CR :package:

**Pull request merged**